apps/prairielearn/src/pages/instructorStudents/instructorStudents.html.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  NuqsAdapter,
+  parseAsColumnPinningState,
+  parseAsColumnVisibilityStateWithColumns,
+  parseAsSortingState,
+} from '../../lib/client/nuqs.js';
+import type { StaffCourseInstanceContext } from '../../lib/client/page-context.js';
+
+import { ColumnManager } from './components/ColumnManager.js';
+import { DownloadButton } from './components/DownloadButton.js';
+import { StudentsTable } from './components/StudentsTable.js';
+import { type StudentRow } from './instructorStudents.shared.js';
+
+// This default must be declared outside the component to ensure referential
+// stability across renders, as `[] !== []` in JavaScript.
+const DEFAULT_SORT: SortingState = [];
+
+const columnHelper = createColumnHelper<StudentRow>();
+
+interface StudentsCardProps {
+  students: StudentRow[];
+  timezone: string;
+  courseInstance: StaffCourseInstanceContext['course_instance'];
+  course: StaffCourseInstanceContext['course'];
+}
+
+function StudentsCard({ students, timezone, courseInstance, course }: StudentsCardProps) {
+  const [globalFilter, setGlobalFilter] = useQueryState('search', parseAsString.withDefault(''));
+  const [sorting, setSorting] = useQueryState<SortingState>(
+    'sort',
+    parseAsSortingState.withDefault(DEFAULT_SORT),
+  );
+  const [columnPinning, setColumnPinning] = useQueryState(
+    'frozen',
+    parseAsColumnPinningState.withDefault({ left: ['user_uid'], right: [] }),
+  );
+
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  // Focus the search input when Ctrl+F is pressed
+  useEffect(() => {
+    function onKeyDown(event: KeyboardEvent) {
+      if ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() === 'f') {
+        if (searchInputRef.current && searchInputRef.current !== document.activeElement) {
+          searchInputRef.current.focus();
+          event.preventDefault();
+        }
+      }
+    }
+    document.addEventListener('keydown', onKeyDown);
+    return () => document.removeEventListener('keydown', onKeyDown);
+  }, []);
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+
+  const columns = useMemo<AccessorKeyColumnDef<StudentRow, any>[]>(
+    () => [
+      columnHelper.accessor('user.uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('user.name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('user.email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('enrollment.created_at', {
+        header: 'Enrolled on',
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return (
+            <FriendlyDate date={date} timezone={timezone} tooltip options={{ includeTz: false }} />
```

Comment:
Discussed on Zoom: let's add and enable an option to remove the display of seconds here.

Reply:
Done in [0255d35](https://github.com/PrairieLearn/PrairieLearn/pull/12211/commits/0255d350a14fe1676ac794423662688406695d95).